### DISCUSSION TITLE
How to check if bucket has object lock enabled or not

### DISCUSSION THREAD

**Discussion by Fais-007** [aws/aws-sdk-java-v2#3463] [Q&A]
I am using this code to check if bucket has object lock enabled or not:

`GetObjectLockConfigurationRequest getObjectLockConfigurationRequest = GetObjectLockConfigurationRequest
				.builder().bucket(bucketName).build();
GetObjectLockConfigurationResponse getObjectLockConfigurationResponse = s3Client
				.getObjectLockConfiguration(getObjectLockConfigurationRequest);
return getObjectLockConfigurationResponse.objectLockConfiguration().objectLockEnabledAsString()
				.equals("Enabled");`

It is working fine for bucket with object lock enabled but throwing exception if object lock is disabled.

software.amazon.awssdk.services.s3.model.S3Exception: Object Lock configuration does not exist for this bucket (Service: S3, Status Code: 404, Request ID: 519BPB5AAMVCEB14, Extended Request ID: 6LK0fOzC7RzVQJK/e7tuW5qVWWg1zWTor9KRf3AOQARA/zrcp+OFGit3Czno2pKzP99l5ko1L7Q=)

Is this the correct way to check for this or is there any other way to do it.

**Comment by debora-ito** [comment]
@Fais-007 apologies for the lack of replies. Looking at the API Reference I couldn't find other API that would return Object Lock properties, so yes calling `GetObjectLockConfiguration` is one way to do it.

**Reply by Fais-007** [reply]
Hi! So if bucket has object lock disabled then this method throws exception. So I can assume that if the method is throwing exception then the bucket has object lock disabled?
